### PR TITLE
Android Build Update

### PR DESCRIPTION
**Pull request by msmalik681**
- set sdk version to 31

- updated gradle from 3.2.1 to 7.0.3

- added update to retain functionality on android 10

- simplified build process on linux and windows.

- added source code fixes to allow android to compile.

- added x64 builds back as they are supported now.

### CODE REVIEW

**Comment by DCurrent** [comment]
I can't merge this due to code conflicts. Make sure you pull down latest global beta and merge into your branch first to make sure there are no conflicts.

File: engine/openbor.c [hunk 1]
```diff
@@ -28504,7 +28504,7 @@ void checkdamageflip(entity* target_entity, entity *other, s_attack *attack_obje
     /* Debuging info */
     //attack_dump_object(attack_object);
 
-    target_entity->normaldamageflipdir = DIRECTION_ADJUST_OPPOSITE;
+    target_entity->normaldamageflipdir = -1;
```
Comment (DCurrent) [inline engine/openbor.c:28507]: This is not an acceptable solution. We do not replace named constants with magic numbers.

Comment (msmalik681) [reply engine/openbor.c:28507]: Sure

Comment (DCurrent) [reply engine/openbor.c:28507]: OK, I'll look into this. For now, leave as is and I'll fix in global_beta.

Comment (msmalik681) [reply engine/openbor.c:28507]: Cool thanks

**Pull request closed without merging**